runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -289,11 +289,16 @@ public void onOverlayMenuClicked(OverlayMenuClicked overlayMenuClicked)
 	@Subscribe
 	public void onMenuOptionClicked(final MenuOptionClicked event)
 	{
-		if (event.getMenuOption() != null && event.getMenuOption().equals("Read"))
+		final boolean isXMarksTheSpotOrb = event.getId() == ItemID.MYSTERIOUS_ORB_23069;
```

Comment:
You should verify the menu type before assuming the id is an item id. I think the old code works because you can getItemComposition() with any id and it always returns a non null item with the name `"null"` if the id isn't valid. However this is bad practice and we should prefer to remove that whenever we see it.